hip_data_tools/apache/cassandra.py

```diff
@@ -221,21 +290,35 @@ def upsert_dataframe(self, dataframe: DataFrame, table: str) -> None:
         """
         prepared_statement = self._session.prepare(
             self._cql_upsert_from_dataframe(dataframe, table))
-        batch = _prepare_batch(prepared_statement,
-                               _extract_rows_from_dataframe(dataframe))
-        return self._session.execute(batch)
 
-    def upsert_dict(self, data: list, table: str) -> ResultSet:
+        batches = _prepare_batches(prepared_statement, _extract_rows_from_dataframe(dataframe))
+        return self._execute_batches(batches)
+
+    def _execute_batches(self, batches):
+        results = []
+        pbar = tqdm.tqdm(total=len(batches))
+        log.info("Executing cassandra batches")
+        for batch in batches:
+            results.append(self._execute_batch(batch))
+            pbar.update(1)
+        log.info("finished %s batches", len(results))
+        return results
+
+    @retry(wait_exponential_multiplier=100, wait_exponential_max=100000)
```

Comment:
should these multiplier and exponential values be configurable from the environment?

Reply:
good point, 
added as env overwritable sensible defaults